File: 백준/Silver/1475. 방 번호/방 번호.py
N = input()
number = [0] * 10
for i in N:
    if int(i) in [6,9]:
        if number[6] < number[9]:
            number[6] += 1
        else:
            number[9] += 1
    else:        
        number[int(i)] += 1

print(max(number))

File: 백준/Silver/1475. 방 번호/방 번호.cc
#include <bits/stdc++.h>
using namespace std;

int n, M;
int arr[10];

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    cin >> n;
    while(n > 0) {
        arr[n%10]++;
        n /= 10;
    }
    
    int n69 = arr[6] + arr[9];
    arr[6] = n69 / 2;
    arr[9] = n69 / 2;
    if (n69%2 == 1) {
        arr[9]++;
    }

    for (int i=0; i<10; i++) {
        M = max(M, arr[i]);
    }
    cout << M;

    return 0;
}